emulator.go

```diff
@@ -37,6 +37,8 @@ type Server struct {
 	tsMux sync.Mutex
 }
 
+var autoCreateQueueOnNewTask = true
```

Comment:
I'm not in favour of global variables if they can be avoided, mainly as it makes testing harder. 
Can we introduce a `ServerOptions` struct (in `Server`) instead that holds this flag?

Also I think it should default to false for consistency.

Reply:
Yes default should be false, I forget to reset it after a test. I added the ServerOptions struct so we avoid global variable.